### DISCUSSION TITLE
Welcome to DtCyber Discussions!

### DISCUSSION THREAD

**Discussion by kej715** [kej715/DtCyber#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by wschaub** [comment]
I'm a retrocomputing hobbyist and life long tinkerer with computers in general.

I've been involved with dtcyber via way of the CYBIS kit since about late 2018/early 2019. Since then I've been trying to learn as much as I can about the internals of NOS and PLATO and wrote a few small patches for both. 

This particular branch I've done some testing of the cartridge tape drive emulation and experimentation with TMS + a patch for the NCC to improve the timing of MAGNET in response to tape mount requests from the tape library and to enable the guest account to mount unlabeled tapes owned by TMS.   I also did a deep dive into IAF and NAM to figure out why timeouts hung instead of logging the user off.

I'm currently exploring the networking features unique to this branch of dtcyber, I have a half finished project doing shared notes files between two CYBIS systems I need to get back to and finish, along with some experimentation with NJEF and RBF. 

Longer term I would like to package up some of the work I've done over the last 3 years and document it well enough that others can dive in and do the same.

**Comment by petermallan** [comment]
My interest in DtCyber goes back to the fact that I did my PhD in
Astrophysics in the late 1970s using a CDC 7600 at the University of
Manchester. (Technically, the computers were part of the University of
Manchester Regional Computing Centre.) I know that DtCyber does not
simulate the 7600, but from a user perspective, NOS 2.8.7 on a Cyber 170 is
not that different to SCOPE 2.1 on a 7600, apart from the fact that it
provides interactive access. Back in my day, it was all punched cards.

Back then I was a language junky and although I used Fortran for my PhD
work, I also learned Pascal, Cobol, Algol 60 and Algol 68, and Compass.
While I learned to do quite a lot as a user, I had no exposure to system
management on the 7600. (Quite right too!)

I later became the system manager of the VAX computers in the astronomy
department for six years, so I am now using DtCyber and NOS 2.8.7 to learn
about system management on such a system. The great thing is that if I mess
it up, I am not inconveniencing anyone else.

The CDC 7600 was part of the so-called Joint System at Manchester. This
consisted of an ICL 1906A running George 3, plus the CDC 7600 running SCOPE
2.1. The main purpose of the ICL 1906A was to act as a front end for the
7600. I am working up to linking together a simulator of the ICL 1900
series computers with DtCyber, but there are a few things I need to find
out first.

Cheers

Peter Allan

**Reply by kej715** [reply]
Hi Peter,

Please keep us updated on your work with George 3. I briefly looked into the possibility of adding an ICL 1906A running George 3 to the collection at the [Nostalgic Computing Center](http://www.nostalgiccomputing.org), but the George 3 emulator didn't appear to have much networking ability when I was looking at it back then. Are you working on adding some type of networking feature to George 3 and/or DtCyber that would enable them to communicate using historical software subsystems (e.g., HASP, NJE, or an ICL-provided RJE protocol)?

cheers,
Kevin

**Comment by petermallan** [comment]
Hi Kevin,

At the moment I am trying to find out how the two computers communicated
'for real'. It could have been a network connection, but I think it was
something more closely linked than that. I do know that other regional
computer centres had a CDC 7600 + Cyber 170 configuration, with the Cyber
being used as the job entry system.

With your knowledge of CDC systems, do you know how a dual 7600 - Cyber 170
(or perhaps some other Cyber system) would likely have talked to each other?

Cheers

Peter

**Reply by kej715** [reply]
Hi Peter,

A number of different options were available for linking Cyber 170's with CDC 7600's. Perhaps the two most common were:

- **RHF/LCN**. LCN (Loosely Coupled Network) was a 50 Mbit/sec networking solution offered by CDC. It supported Cyber 170's, 7600's, 200's, and a variety of non-CDC systems. It had a bus-type of architecture to which multiple computer systems could be connected concurrently, resembling ethernet in some ways, although it was significantly faster than ethernet during that era.
- **SCOPE Station**. This was a type of channel-to-channel solution for connecting 170's to 7600's.

Possibly, RJE-based solutions were available too. They would be very slow compared to RHF/LCN and SCOPE Station, so probably not used for interconnecting 170's to 7600's, but might be what was used to connect George 3 to 7600's. Network Systems Corp Hyperchannel probably supported 7600's with SCOPE as well, and it's conceivable that they would have supported ICL and George 3 too.

cheers,
Kevin

**Comment by petermallan** [comment]
Thanks Kevin. That lets me ask intelligent questions of a few people who
might know the answer.

Happy New Year

Peter

**Comment by petermallan** [comment]
Hi Kevin.

I have thought of something more that might be relevant to the question of
how the ICL 1906A and the CDC 7600 were linked. In order to access a file
that was physically held on the 1906A, the relevant SCOPE statement was
something like

ATTACH(FILE1,permanent_file_name,ST=S6A)

Does the ST=S6A part imply anything about whether LCN, SCOPE station, or a
network link was the physical means of connection?

Cheers

Peter

**Comment by kej715** [comment]
Hi Peter,

SCOPE Station predated RHF/LCN.  The "ST" parameter identifies a "station"
and was introduced in SCOPE 2 for SCOPE Station. RHF/LCN subsequently
adopted the "ST" parameter as a more generalized identifier for nodes in an
LCN network. Later, other applications on NOS adopted the concept too,
including TLF, NJF, and Cray Station.

I didn't have an opportunity to use or work on a 7600 or SCOPE 2, and I
don't know how RHF was exposed to users of the 7600. In particular, I don't
know whether the ATTACH and GETPF commands were adapted to use RHF/LCN in
addition to SCOPE Station, or instead something like the NOS MFLINK command
was added to SCOPE for RHF/LCN. My bet would be on the former. In that
case, the ATTACH command you've referenced could have been supported by
either technology. However, supporting a link to an ICL machine via LCN/RHF
probably would have required that CDC provided a type of NAD (LCN Network
Access Device) for  the ICL machine, and I don't recall seeing ICL on any
list of CDC-supported NAD types. It might have been easier for ICL to
implement a channel adapter to the 7600. That would have enabled
implementing the SCOPE Station interface.

cheers,
Kevin

**Comment by petermallan** [comment]
Thanks Kevin. That is very useful information.

Cheers

Peter

**Comment by rly12001** [comment]
Hi Kevin:

Your mods to dtcyber are very impressive. Here are a couple of items that you may want consider:

- since you have added the expect script to build and configure NOS, how about having the ability to specify configuration values (i.e. the iP numbers, hostnames, etc). Something like a rc file that can include products configuration info. I would use this to configure stk and ip networking (my home network IPs conflict with NCCs.

- Could you provide a small write-up on how to configure the tape library contents for stk/atf/et al. Specifically defining a scratch pool and how to import "external" tapes to the library.

thanks
-ron

**Reply by KC5VDJ** [reply]
Yeah, I may play with the reconfigure script to add other decks.  We'll see, gotta find one of those round tuits.  For cmrd, eqpd, and network, it seems to do well enough, but I guess it was never finished.  Also options for non-default passwords as well during reconfig would probably make everyone happy too.  Still trying to figure out the octopus called modify.  I guess the major advances of codebase management came in the 90's...still, for what it did...

**Comment by kej715** [comment]
Hi Ron,

Thanks for reaching out and providing some great feedback. I really like
your "rc" idea. A major goal has been to make the installation of NOS as
automated as possible, with as little need as possible to require any info
from the user to accomplish the install. However, your "rc" idea strikes a
very nice balance. If no "rc" file exists, the process could behave as it
currently does. If the optional "rc" is detected, it could be read to
obtain user-supplied preferences.

Regarding lack of doc on the Stk simulator ... this is a well-known (and
somewhat embarrassing) deficiency that's high on the to-do list to resolve.
Stay tuned ... the repo is changing frequently, with multiple pushes
tending to occur daily.  Regarding importing "external" tapes, use the
"path" attribute in the "volumes.json" file instead of the "url" attribute,
and set the value of the "path" attribute to the pathname of a standard
".tap" file.

thanks again!
Kevin

**Comment by rly12001** [comment]
I'm eagerly following your changes... one quick question: Is the file "./NOS2.8.7/files/TMS-catalog.txt missing? It is referenced in ./NOS2.8.7/decks/load-tms-catalog.job.

thanks
-ron

**Comment by kej715** [comment]
Hi Ron,

The script "./NOS2.8.7/activate-tms.exp" calls "curl" to query the
StorageTek 4400 web service for information about all of the volumes that
the StorageTek 4400 knows about, and it requests that the web service
produces a format that can be fed to TFSP as its input file. The output
from the web service is stored in "./NOS2.8.7/files/TMS-catalog.txt", and
then "./NOS2.8.7/decks/load-tms-catalog.job" is loaded on the card reader.
As a result, the TMS catalog is initialized to reflect the tape volumes
known to the Stk simulator. Thus, "./NOS2.8.7/files/TMS-catalog.txt" is
created dynamically, as needed, as part of a strategy for automating the
synchronization of the TMS catalog with the Stk server's volume
configuration.

cheers,
Kevin

**Comment by rly12001** [comment]
Hi Kevin:

one minor issue... I rebuilt everything from scratch this morning. Everything went well, except for building ncctcp. It aborted with an invalid command "CYBCCMN". I have not yet built a new deadstart tape, so I'm still running on the original one.

Here is ncctcp's dayfile:
12.21.37.NCCTCP. 
12.21.37.UCCR, 01, 020,      0.033KCDS.
12.21.37.$USER,INSTALL,.       
12.21.37.ABSC,  B. 
12.21.37.$NORERUN. 
12.21.37.$SETTL,*. 
12.21.37. TL = UNLIMITED.
12.21.37.$SETASL,*.
12.21.37. ASL = UNLIMITED, JSL = UNLIMITED.
12.21.37.$SETJSL,*.
12.21.37. ASL = UNLIMITED, JSL = UNLIMITED.
12.21.37.$ASSIGN,51,TAPE,D=GE,LB=KU,F=I,PO=R.
12.21.38. NT051 ASSIGNED TO TAPE, VSN=***051.
12.21.38.$COPYBF,TAPE,TCPMAKE. 
12.21.38. COPY COMPLETE. 
12.21.38. END. 1 FILE; 10 RECORDS; 357 WORDS.
12.21.38.$REPLACE,TCPMAKE. 
12.21.38.$PURGE,TCPBASE/NA.
12.21.38.$DEFINE,TCPBASE.
12.21.38.$COPYBF,TAPE,TCPBASE. 
12.21.39. COPY COMPLETE. 
12.21.39. END. 1 FILE; 34 RECORDS; 83712 WORDS.
12.21.39.$RETURN,TCPBASE.
12.21.39.TCPMAKE.
12.21.39.BEGIN,HTF,TCPMAKE.
12.21.39.IF,.NOT.FILE(CCTCPIO,AS),CCTCPIO. 
12.21.39.  BEGIN,CCTCPIO,TCPMAKE.
12.21.39.RETURN,COMPILE,LGO,CCTCPIO. 
12.21.39.SES.GENCOMP M=(ZN7IJOB,ZN7IQTI,ZN7IUTL) B=TCPBASE 
12.21.39.CYBCCMN 
12.21.39. INCORRECT COMMAND. 
12.21.39.$EXIT.CCL         
12.21.40.$REVERT,ABORT.CCL 
12.21.40.$EXIT.CCL         
12.21.40.$REVERT,ABORT.CCL 
12.21.40.$EXIT.CCL         
12.21.40.$REVERT,ABORT.CCL 
12.21.40.EXIT. 
12.21.40.*** 
12.21.40.*** NCCTCP FAILED 
12.21.40.*** 
12.21.40.UEAD,      0.002KUNS. 
12.21.40.UEPF,      0.034KUNS. 
12.21.40.UEMT,      0.780KUNS. 
12.21.40.UEMS,      6.951KUNS. 
12.21.40.UECP,      0.064SECS. 
12.21.40.AESR,      3.908UNTS. 
12.21.40.$OUT(*/OP=E)
12.21.40. NO FILES PROCESSED.
12.21.40.$UNLOAD(*/OP=O) 
12.21.40. NO FILES PROCESSED.
12.21.40.$DAYFILE(OUTPUT,JT=D) 
12.21.41.UCLP, 01, 021,      0.128KLNS.

**Comment by rly12001** [comment]
ncctcp appears to build correctly after using the newly built deadstart tape.

thanks
-ron

**Comment by kej715** [comment]
Hi Ron,

Yes, ncctcp depends upon ses, and that is intended to imply that SES must be fully installed which, in turn, means it’s on the deadstart tape or has been SYSEDIT’ed into the running system.

I’ll update the README.md to clarify this.

Cheers,
Kevin

**Comment by kej715** [comment]
On second thought, William Schaub suggested a way to avoid the requirement to create a new deadstart tape or do a SYSEDIT, so I’ll update the ncctcp job to implement his suggestion. :-)

Cheers,
Kevin

**Comment by rly12001** [comment]
Hi:

with the latest version of dtcyber, expect prep-customization.exp aborts with a missing file.

ron@ryzen:~/cyber-new/kevin/DtCyber/NOS2.8.7$ expect prep-customization.exp 
spawn nc localhost 6666

Operator> spawn tail -F LP5xx_C12_E5
e [UNLOAD,51.

Operator> e [UNLOAD,52.

Operator> e [UNLOAD,53.

Operator> lt 13,0,1,r,tapes/nos287-1.tap
Successfully loaded tapes/nos287-1.tap

Operator> lt 13,0,2,r,tapes/nos287-2.tap
Successfully loaded tapes/nos287-2.tap

Operator> lt 13,0,3,r,tapes/nos287-3.tap
Successfully loaded tapes/nos287-3.tap

Operator> e [X.SYSGEN(SOURCE)

Operator> 
SYSGEN started
SYSGEN complete
lc 12,4,decks/init-build.job
Cards loaded on card reader C12,E4

Operator> 
IBUILD started
IBUILD complete
lc 12,4,decks/modopl.job
Failed to open /home/ron/cyber-new/kevin/DtCyber/NOS2.8.7/decks/../mods/NOS/TMSNOPV.mod

Operator> 
***** Error condition encountered *****
Trying to execute load_job 12 4
Entering interact mode. Type interp to get an interpreter.

**Comment by kej715** [comment]
Argh!  Sorry about that. The missing file has been committed and pushed to
the repo.

**Comment by rly12001** [comment]
Hi Kevin:
my email is getting rejected again (is there anyway that you can add me to a whitelist?). Here is more information:

> Normally, TMS is turned off for SYOT jobs, and DIS runs as an SYOT job. If
> you want to use TMS-managed tapes from DIS, you'll need to execute the
> following command first, after entering DIS:
>
>     TMSDEF,TO=3DT.
>
> or, add the TO=3DT parameter to the LABEL command.
>
> cheers,
> Kevin
>

	still no joy... the only difference is that is says "waiting for 
	vsn=msua00, at". It used to say ge.

	I used wireshark and turned debugging on for stk...
	I see packets to/from port 4400 at deadstart and then nothing
	when I issue the label command.
	E,P doesn't show any requests, E,T only have the deadstart tape on
	NT050 mounted.

	Init is running on port 111, stk is trying to bind to it. Could this
	be the problem? I am running fuse filesystems and nfs on this machine.

	thanks
	-ron

	Here is the stk.log file. Note I do not have ipv6 enabled, ipv4 only.

> stk@1.0.0 start
> node server.js

3/25/2022, 1:55:34 PM PID is 151684
3/25/2022, 1:55:34 PM HTTP server listening on port 4480
3/25/2022, 1:55:34 PM Error: bind EACCES 0.0.0.0:111
    at node:dgram:351:20
    at processTicksAndRejections (node:internal/process/task_queues:84:21)
3/25/2022, 1:55:34 PM RPC prog 300031 vers 1 listening on 0.0.0.0:4400
3/25/2022, 1:55:34 PM Tape library located at /home/ron/cyber-new/kevin/DtCyber/
stk/tapes
3/25/2022, 1:55:34 PM Tape cache located at /home/ron/cyber-new/kevin/DtCyber/st
k/tapes/cache
3/25/2022, 1:55:34 PM Tape server listening on port 4400
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:42996 connected
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:42998 connected
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:43000 connected
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:43002 connected
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:42996 <= REGISTER M0P0D0
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:42996 => 200 M0P0D0 registered
3/25/2022, 1:56:11 PM M0P0D0 registered by ::ffff:127.0.0.1
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:42998 <= REGISTER M0P0D1
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:42998 => 200 M0P0D1 registered
3/25/2022, 1:56:11 PM M0P0D1 registered by ::ffff:127.0.0.1
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:43000 <= REGISTER M0P0D2
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:43000 => 200 M0P0D2 registered
3/25/2022, 1:56:11 PM M0P0D2 registered by ::ffff:127.0.0.1
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:43002 <= REGISTER M0P0D3
3/25/2022, 1:56:11 PM StkCSI TCP ::ffff:127.0.0.1:43002 => 200 M0P0D3 registered
3/25/2022, 1:56:11 PM M0P0D3 registered by ::ffff:127.0.0.1

**Comment by Hal-Barnes** [comment]
Hello Everyone -


Hal Barnes here, CDC employee from 87-92, also a CYBER user / student operator in my college days, 82-87.

I've seen references to DtCyber over the years and always wanted to try it out, but in the past, BT had not yet released NOS (the OS I'm interested in) to the public so there wasn't much motivation.  It's really exciting to see all the activity now, just wish I'd found this sooner!

I'm primarily interested in learning more about NOS 2.8.7 operations and actually taking a few PLATO/CYBIS courses.  As a student operator, I had access to the console but couldn't play much as it was a production system; DtCyber is perfect for this.  Thanks to all the people who created, maintained and expanded the simulator over the years - your hard work is greatly appreciated!


So.  A couple of weeks ago, I downloaded the source from github and attempted a compile.  The project wanted C build tools 2015 which I installed; the build failed with a link error about _dsa311Init.

When I download today, the build fails with the following:

1>device.obj : error LNK2001: unresolved external symbol _msufrendInit
1>device.obj : error LNK2001: unresolved external symbol _dsa311Init

I'm using Windows 11, VS2019 and on another machine, VS2022; same error on both.  Probably something really simple but it's beyond my abilities to resolve.



For me, the CYBIS build located here - along with the DtCyber 5.5.1.SZ binary - have been the most useful:

https://codex.sjzoppi.com/doku.php?id=utilities:start

Also, I was able to download the source from the CODEX and compile it successfully.  The source says it is the Tom Hunter base of 5.5.1, but it has a way to change the screen size (in const.h) which per the text on the page was added later by Steve Zoppi as 5.5.1.SZ so I'm a little confused by that.  The screen sizing feature is important for me because while various DtCyber binaries work great on an external screen, on my Surface 8 laptop, I need the smaller screen to make it useful.  The auto paper unload and ability to specify I/O folders in the .SZ version are nice to haves, too.  Are there any plans to incorporate those features into this trunk?  Having the screen size set in the INI file would be a really nice touch, too.

The coding involved in most of this is beyond me, but I'd like to dig in and help out where I can.

Thanks!
Hal

**Comment by kej715** [comment]
Hi Hal,

The build errors you're observing are likely due to the files msufrend.c,
msufrend_util.c, and dsa311.c being missing from your Visual C/C++ project
database. Try adding them to your project and then re-building.

cheers,
Kevin

**Comment by kennethfugate** [comment]
Is there an alternate way to send control-T to exit TEXT mode in NOS2.8.7 IAF?  When I press control-T, web browsers want to open a new tab.  I see that it is listed as \x14 in the text scripts.  Is it an alt keystroke sequence?

I've started using FSE but would like to also be able to use TEXT mode.

Ken

**Comment by kej715** [comment]
Hi Ken,

You can press control-C followed by Enter instead of pressing control-T.
Also, Firefox does not recognize control-shift-T as a keyboard shortcut, so
if you're a Firefox user, you can use control-shift-T instead of control-T.
Browsers (e.g., Firefox, Chrome, Edge) that use control-T as a keyboard
shortcut to open a new tab don't seem to provide a way to override this
behavior (and overriding might confuse users anyway).

cheers,
Kevin

**Comment by KC5VDJ** [comment]
Just popping in to say hi!   I've checked out dtcyber off and on for geez, I forget, when it appeared in the freebsd ports collection, I guess.  But it took the generosity of BT to get me doing much with it with their release of NOS and pretty much most of the tools from what I can see (a complete set of manuals would be nice...too many currently unavailable or obscurely located (buried in archive.org maybe?)..I digress).

My Cyber experience comes by way of the USN's PMTC, Point Mugu, where I was the clerk-trainee, then computer clerk in the systems analysis section, basically the gopher, pc assembly and c, ftn5, etc.  Started as the tape librarian, and yes, I'd like to get tms working 😉.

It was a cool job, with a crew of cool people, at a cool time, when (love him, hate him, whatevuh), when "RayGun" used to park next to the MAD in VXE-6's hangar.  All the f'ing time it seemed.  On a trippy side note, a future supervisor at the sprint ldd Lenexa ossc where I was an HP-UX, Solaris and VMS systems manager was a TSgt in the comms room on his plane when I was working at PMTC.

Anyhow, after decades of everything else, full circle!

Has anyone worked towards ax.25 support?

LTP?  How many CDC computers did he buy?

I will be posting the options for the Intel backend for vs22.  They seem to work, with ipo.  Still trying to find the bounds checking wrapper lib for that option though, until then....  Microsoft vs is a great ide, but their code generation has always sucked, and it looks like Intel may be free now.  2x to 5x improvement on most computational tasks simply by recompiling in most cases over Microsoft.  Seamless HPF integration too.

**Reply by KC5VDJ** [reply]
Raise your hand if you can break any MS-DOS MS-C (any version) with a single line program resulting in "RegMD Error..please call Microsoft technical support"..  Good for free upgrades back in the day.

**Comment by philclaridge** [comment]
Try here for manuals: http://www.bitsavers.org/pdf/cdc/Tom_Hunter_Scans/

See also:
http://www.nostalgiccomputing.org <http://www.nostalgiccomputing.org/index.html> - login to CDC
https://cdc.sjzoppi.com/ - lots of CDC links

Phil

**Reply by wschaub** [reply]
The entire pdf/cdc folder is full of great material on bitsavers, not just the tom hunter scans folder.  the cdc.sjzoppi.com link mentioned above also has a great deal of software in tape image form as well has a lot of good articles and research.  

I've found the bitsavers collection to be pretty comprehensive and well worth the effort to scour through.  If there's anything anyone is looking for I more than likely know where to find it assuming it's been scanned.

**Reply by KC5VDJ** [reply]
Thanks, I have those, but there are some missing.  I have done searches, and some seem to be cataloged in boxes at the computer history museum, awaiting a volunteer to scan them, I presume?  Too damn broke for that here.

**Reply by wschaub** [reply]
Do you have a list? I think sometimes Al Kossow will take requests to scan in manuals.

**Reply by KC5VDJ** [reply]
Hmmm, I'll have to dig up the search history and make one.  Iirc, it was the proper UT-Lisp manual and some others.  Other than that, the best I can find is the 1975 manual for UT-Lisp, and one from iirc, 1968.

**Comment by kznpoike** [comment]
Hello All,

To introduce myself, my name is John Huntley. I worked at ESKOM, the South African power utility, from 1979 to 1986, mostly on a Cyber 175. I was in the system support team so did NOS O/S support as well as CCP support on some custom communication software we had linking up to ICL 1900 and 2900 systems. In 1986 I joined CDC professional services in Copenhagen. Worked mostly on NOS/VE there as I had spent 9 months with the NOS/VE development team in Arden Hills. My last CDC spell was 1991 to 1993 in Arden Hills in the performance team, by then working mainly on 4000 series EPIC systems. I was the sad one who put the lights out on the performance team at the end of 1992 :-(.

Now I am retired and my interest in DtCyber is nostalgic. Finding out how little of NOS I remember, although it is slowly coming back. I have been picking at a native Wayland implementation of the local console Window_xx module. I have a first cut running on a Raspberry Pi 3B that has nothing else to do. If anybody is interested please let me know.

**Comment by kej715** [comment]
Hi John,

Welcome to DtCyber discussions. Our times in Arden Hills overlapped. I
worked there from 1988 to 1998, as a member of the CDCNet Design team, and
then in EP/IX Development. We probably passed each other in the corridors
and sat in many of the same all-hands meetings in the cafeteria.

cheers,
Kevin

**Reply by kznpoike** [reply]
Hi Kevin,

A faint bell went off in my mind when I saw your name but was not sure. Nice to touch base again.

**Comment by philclaridge** [comment]
Hello John,
Welcome and hello from Cambridge, UK.

Would love to have a chat about the work you are doing.

I am finishing off the my generation character textures (basically just a set of PNG of all of the characters in various sizes and weights, replacing the X11 fonts).  First generation is just fixed width lines, in the second generation textures (later this year) you will be able to dial in some real-world distortion (XY drive amplifiers, beam velocity) more like a real dd60. These textures should be usable in any of the newer desktop or web displays. Let me know if interested.

Are you using the direct wayland_client or SDL3 (the latter is the direction I was thinking about when textures were done as I can then use Pi, Linux, and Mac).

Phil Claridge

**Reply by kznpoike** [reply]
Hello Phil,

The code I have so far is a direct Wayland client using FontConfig and FreeText for character rendering. A couple of reasons, not the least that I have never written client side graphics code like this before, so have a reasonable learning curve. It is enough to learn the basics let alone other layers, also I have only been thinking about Unix style systems. I also vaguely remembered reading that somebody was looking into emulating the "real" character set and thought that keeping a vanilla pixel image might enable something.

The starting point for me was trying the emulator on an old Raspberry Pi 3B, the original 1200MHz one. Raspberrypi O/S has now updated to native Wayland, so the X11 code is running through the XWayland emulation and the console was basically unusable. The NOS real time clock managed to update every 10 - 15 seconds and keyboard response around 30 seconds. So that started me thinking about wasting some time learning Wayland :-). My current code is still quite rough but functional. On the same Raspberry PI system the real time clock keeps up, dynamic displays are quite lively and the keyboard response in the 1 seconds range, so all around usable. I do have one problem, not knowing what the screen clearing protocols are for a 6612 or CC545 I decided to do a "fade" like a real CRT tube will do, every frame refresh I divide the pixel intensity by 2. This provides a good visual effect, but makes the display sensitive to PP performance. When the load starts to pressure the PP's the screen blanks out. I sort of remember behavior like this on a real, highly overloaded, 175 in the old days. I think that it should be quite straight forward to change the code that paints the pixels in the image map so doing some prototyping should not be too difficult.

The code is pushed into my fork in the main branch if you want to look at it.

I would love to explore things with you. We should probably move to some other discussion thread. I am not hardwired into anything I have done and another path might well be better. All I will say from my experience so far is that you need a layer that ultimately comes down to a direct Wayland compositor base with reasonable efficiency.

Regards,
John Huntley.
